sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1996,7 +1996,7 @@ class Dataset[T] private[sql](
   def dropDuplicates(colNames: Seq[String]): Dataset[T] = withTypedPlan {
```

Comment:
You have to add more documentation for streaming usage! especially you have to document that this will keep all past data as intermediate state, and you can use the `withWatermark` to limit how late the duplicate data can be and system will accordingly limit the state.

Also, double the docs on withWatermark and make sure its consistent.